enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,335 @@
+2. Embed this Ignition to an RHCOS liveCD.
+3. Boot a machine with this liveCD.
+
+This POC uses the following services for mitigating some gaps:
+- `patch.service` for allowing single node installation. it won't be required once [single-node production deployment](https://github.com/openshift/enhancements/pull/560/files) is implemented.
+- `post_reboot.service` for approving the node CSR and bootstrap static pods resources cleanup post reboot.
+
+Steps to try it out:
+- Clone the installer branch: `iBIP_4_6` from https://github.com/eranco74/installer.git
+- Build the installer (`TAGS=libvirt hack/build.sh`)
+- Add your ssh key and pull secret to the `./install-config.yaml`
+- Generate ignition - `make generate`
+- Set up networking - `make network` (provides DNS for `Cluster name: test-cluster, Base DNS: redhat.com`)
+- Download rhcos image - `make embed` (download RHCOS liveCD and embed the bootstrap Ignition)
+- Spin up a VM with the the liveCD - `make start-iso`
+- Monitor the progress using `make ssh` and `journalctl -f -u bootkube.service` or `kubectl --kubeconfig ./mydir/auth/kubeconfig get clusterversion`
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+## Design Details
+
+### Open Questions
+
+1. How will the user specify custom configuration, such as installation disk, static IPs?
+2. Number of revisions for the control plane - do we want to make changes to the bootstrap static pods to make them closer to the final ones?
+3. What do we do with the CRDs that are generated by operators instead of installed from manifests managed by the CVO?
+
+### Bootable installation artifact (future work)
+
+In order to embed the bootstrap-in-place-for-live-iso Ignition config to the liveCD the user need to get the liveCD and the coreos-installer binary.
+We consider adding `openshift-install create single-node-iso` command that that result a liveCD with the bootstrap-in-place-for-live-iso.ign embeded.
+It can also take things like additional manifests for setting the RT kernel (and kernel args) via MachineConfig as well
+ as supporting injecting network configuration as files and choosing the target disk drive for coreos-installer.
+Internally, create single-node-iso would compile a single-node-iso-target.yaml into Ignition (much like coreos/fcct)
+ and include it along with the Ignition it generates and embed it into the ISO.
+
+### Limitations
+
+While most CRDs get created by CVO some CRDs are created by the operators, since during the bootstrap phase there is no schedulable node, 
+ operators can't run, these CRDs won't be created until the node pivot to become the master node. 
+This imposes a limitation on the user when specifying custom manifests prior to the installation. 
```

Comment:
We should be clear about what that limitation is. Using these CRDs may block bootstrapping and prevent the installation from completing, right?

Reply:
yes